roles/booted_deployment_set_fact/tasks/main.yml

```diff
@@ -3,30 +3,54 @@
 #
 #  This role sets facts for the current booted deployment
 #
+#  Optionally, you can archive the values to an YAML file that will drop into
+#  the current playbook directory.
+#
+- name: Set archive value
+  set_fact:
+    bdsf_archive: "{{ archive | default('false') }}"
+
 - name: Get rpm-ostree status output
   command: rpm-ostree status --json
-  register: ros_status
+  register: bdsf_ros_status
 
 - name: Set json output
   set_fact:
-     ros_json: "{{ ros_status.stdout | from_json }}"
+     bdsf_ros_json: "{{ bdsf_ros_status.stdout | from_json }}"
 
 - name: Set ros variable if deployment 0 is booted
   set_fact:
-    ros_booted: "{{ ros_json['deployments'][0] }}"
-    ros_not_booted: "{{ ros_json['deployments'][1] if ros_json['deployments'][1] is defined else false }}"
-  when: ros_json['deployments'][0] is defined and ros_json['deployments'][0]['booted']
+    bdsf_ros_booted: "{{ bdsf_ros_json['deployments'][0] }}"
+    bdsf_ros_not_booted: "{{ bdsf_ros_json['deployments'][1] if bdsf_ros_json['deployments'][1] is defined else false }}"
+  when: bdsf_ros_json['deployments'][0] is defined and bdsf_ros_json['deployments'][0]['booted']
 
 - name: Set ros variable if deployment 1 is booted
   set_fact:
-    ros_booted: "{{ ros_json['deployments'][1] }}"
-    ros_not_booted: "{{ ros_json['deployments'][0] if ros_json['deployments'][0] is defined else false }}"
-  when: ros_json['deployments'][1] is defined and ros_json['deployments'][1]['booted']
+    bdsf_ros_booted: "{{ bdsf_ros_json['deployments'][1] }}"
+    bsdf_ros_not_booted: "{{ bdsf_ros_json['deployments'][0] if bdsf_ros_json['deployments'][0] is defined else false }}"
+  when: bdsf_ros_json['deployments'][1] is defined and bdsf_ros_json['deployments'][1]['booted']
```

Comment:
Sorry one more suggestion.  I filed #230 to update the rpm_ostree_status and rpm_ostree status_verify to use json_query.  I think that would cleanup this.  Since we only care about the booted deployment, we can do

```
- name: Set booted deployment
  set_fact:
    bdsf_ros_booted: "{{ item }}"
  with_items: "{{ bdsf_ros_json | json_query(query) }}"
  vars:
    query: "[?booted=='true']"
```
Untested so the query might need some work since I don't know if it respects boolean values or reads them as strings.  I believe everything below should work.